Tried with cirq v0.7.0, and got following failures.

```
________________________ test_serialize_sympy_constants ________________________

    def test_serialize_sympy_constants():
        proto = _arg_to_proto(sympy.pi, arg_function_language='')
        packed = json_format.MessageToDict(proto,
                                           including_default_value_fields=True,
                                           preserving_proto_field_name=True,
                                           use_integers_for_enums=True)
>       assert packed == {'arg_value': {'float_value': float(np.float32(sympy.pi))}}
E       AssertionError: assert {'arg_value':...592653589793}} == {'arg_value':...927410125732}}
E         Differing items:
E         {'arg_value': {'float_value': 3.141592653589793}} != {'arg_value': {'float_value': 3.1415927410125732}}
E         Use -v to get the full diff

cirq/google/arg_func_langs_test.py:108: AssertionError
___________________ test_serialize_conversion[value1-proto1] ___________________

value = array([ True, False])
proto = {'arg_value': {'bool_values': {'values': [True, False]}}}

    @pytest.mark.parametrize('value,proto', [
        ((True, False), {
            'arg_value': {
                'bool_values': {
                    'values': [True, False]
                }
            }
        }),
        (np.array([True, False], dtype=np.bool), {
            'arg_value': {
                'bool_values': {
                    'values': [True, False]
                }
            }
        }),
    ])
    def test_serialize_conversion(value: ARG_LIKE, proto: v2.program_pb2.Arg):
        msg = v2.program_pb2.Arg()
        json_format.ParseDict(proto, msg)
        packed = json_format.MessageToDict(_arg_to_proto(value,
>                                                        arg_function_language=''),
                                           including_default_value_fields=True,
                                           preserving_proto_field_name=True,
                                           use_integers_for_enums=True)

cirq/google/arg_func_langs_test.py:138: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cirq/google/arg_func_langs.py:131: in _arg_to_proto
    msg.arg_value.bool_values.values.extend(value)
/nix/store/86g6marsgjjr7yqs903rzp0dynxbcck0-python3.7-protobuf-3.8.0/lib/python3.7/site-packages/google/protobuf/internal/containers.py:281: in extend
    new_values = [self._type_checker.CheckValue(elem) for elem in elem_seq_iter]
/nix/store/86g6marsgjjr7yqs903rzp0dynxbcck0-python3.7-protobuf-3.8.0/lib/python3.7/site-packages/google/protobuf/internal/containers.py:281: in <listcomp>
    new_values = [self._type_checker.CheckValue(elem) for elem in elem_seq_iter]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <google.protobuf.internal.type_checkers.TypeCheckerWithDefault object at 0x7fffd776ad10>
proposed_value = True

    def CheckValue(self, proposed_value):
      """Type check the provided value and return it.
    
      The returned value might have been normalized to another type.
      """
      if not isinstance(proposed_value, self._acceptable_types):
        message = ('%.1024r has type %s, but expected one of: %s' %
                   (proposed_value, type(proposed_value), self._acceptable_types))
>       raise TypeError(message)
E       TypeError: True has type <class 'numpy.bool_'>, but expected one of: (<class 'bool'>, <class 'numbers.Integral'>)

/nix/store/86g6marsgjjr7yqs903rzp0dynxbcck0-python3.7-protobuf-3.8.0/lib/python3.7/site-packages/google/protobuf/internal/type_checkers.py:109: TypeError
______________________________ test_async_sample _______________________________

pyfuncitem = <Function test_async_sample>

    def pytest_pyfunc_call(pyfuncitem):
        if inspect.iscoroutinefunction(pyfuncitem._obj):
            # coverage: ignore
>           raise ValueError(f'{pyfuncitem._obj.__name__} is async but not '
                             f'decorated with "@pytest.mark.asyncio".')
E           ValueError: test_async_sample is async but not decorated with "@pytest.mark.asyncio".

cirq/conftest.py:14: ValueError
_____________________________ test_asyncio_pending _____________________________
[...] MORE of similar error as above
```

--- a/pkgs/development/python-modules/cirq/default.nix
+++ b/pkgs/development/python-modules/cirq/default.nix
@@ -1,0 +1,86 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, google_api_python_client
+, matplotlib
+, networkx
+, numpy
+, pandas
+, pythonProtobuf  # pythonPackages.protobuf
+, requests
+, scipy
+, sortedcontainers
+, sympy
+, typing-extensions
+  # test inputs
+, pytestCheckHook
+, pytest-benchmark
+, ply
+, pydot
+, pyyaml
+, pygraphviz
+}:
+
+buildPythonPackage rec {
+  pname = "cirq";
+  version = "0.6.1";
+
+  disabled = pythonOlder "3.5";
+
+  src = fetchFromGitHub {
+    owner = "quantumlib";
+    repo = "cirq";
+    rev = "v${version}";
+    sha256 = "0lhr2dka7vpz9xd6akxphrcv2b3ni2cgjywpc1r7qpqa5mrq1q7f";
+  };
+
+  # Cirq 0.6 requires networkx==2.3 only for optional qiskit dependency/test, disable this to avoid networkx version conflicts. https://github.com/quantumlib/Cirq/issues/2368
+  # Cirq locks protobuf==3.8.0, but tested working with default pythonPackages.protobuf (3.7). This avoids overrides/pythonPackages.protobuf conflicts
+  prePatch = ''
+    substituteInPlace requirements.txt --replace "networkx==2.3" "networkx" \
+      --replace "protobuf==3.8.0" "protobuf"
+
+    # Fix sympy 1.5 test failures. Should be fixed in v0.7
+    substituteInPlace cirq/optimizers/eject_phased_paulis_test.py --replace "phase_exponent=0.125 + x / 8" "phase_exponent=0.125 + x * 0.125"
+    substituteInPlace cirq/contrib/quirk/cells/parse_test.py --replace "parse_formula('5t') == 5 * t" "parse_formula('5t') == 5.0 * t"
+  '';
+
+  propagatedBuildInputs = [
+    google_api_python_client
+    numpy
+    matplotlib
+    networkx
+    pandas
+    pythonProtobuf
+    requests
+    scipy
+    sortedcontainers
+    sympy
+    typing-extensions
+  ];
+
+  doCheck = true;
+  # pythonImportsCheck = [ "cirq" "cirq.Ciruit" ];  # cirq's importlib hook doesn't work here
+  dontUseSetuptoolsCheck = true;
+  checkInputs = [
+    pytestCheckHook
+    pytest-benchmark
+    ply
+    pydot
+    pyyaml
+    pygraphviz
+  ];
+  # TODO: enable op_serializer_test. Error is type checking, for some reason wants bool instead of numpy.bool_. Not sure if protobuf or internal issue
+  pytestFlagsArray = [
+    "--ignore=dev_tools"  # Only needed when developing new code, which is out-of-scope
+    "--ignore=cirq/google/op_serializer_test.py"  # investigating in https://github.com/quantumlib/Cirq/issues/2727
+  ];
+
+  meta = with lib; {
+    description = "A framework for creating, editing, and invoking Noisy Intermediate Scale Quantum (NISQ) circuits.";
+    homepage = "http://github.com/quantumlib/cirq";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ drewrisinger ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -1824,6 +1824,10 @@ in {
 
   cmarkgfm = callPackage ../development/python-modules/cmarkgfm { };
 
+  cirq = callPackage ../development/python-modules/cirq {
+    pythonProtobuf = self.protobuf;
+  };
+
   colorcet = callPackage ../development/python-modules/colorcet { };
 
   coloredlogs = callPackage ../development/python-modules/coloredlogs { };

